### PR TITLE
util/parse - Additional parsing functions.

### PR DESCRIPTION
**Pull request by ccollins476ad**
* Allow base to be specified for byte streams.
* Parsing function for IPv6 network string.

### CODE REVIEW

**Comment by sterlinghughes** [comment]
@mkiiskila can you look at these, and if good merge?

**Comment by ccollins476ad** [comment]
FYI- there was a bug in parsing address strings with leading double colons (e.g., "::1:2:3:4").  I just force pushed a fix.

**Comment by sterlinghughes** [comment]
@mkiiskila can you review?

File: net/ip/mn_socket/include/mn_socket/mn_socket.h [hunk 1]
```diff
@@ -187,6 +187,9 @@ int mn_close(struct mn_socket *);
 /*
  * Address conversion
  */
+
+#define MN_INET6_MAX_LEN    (39)    /* Does not include terminating \0. */
```
Comment (mkiiskila) [inline net/ip/mn_socket/include/mn_socket/mn_socket.h]: To match the naming of the other items should use the following instead. Also note the size. And if adding this one, probably should add the MN_INET_ADDRSTRLEN there as well :)

#define MN_INET6_ADDRSTRLEN        46

File: util/parse/src/parse.c [hunk 2]
```diff
@@ -261,3 +190,67 @@ parse_byte_stream_exact_length(const char *sval, uint8_t *dst, int len)
...
+int
+parse_byte_stream_exact_length(const char *sval, uint8_t *dst, int len)
+{
+    return parse_byte_stream_exact_length_base(sval, 0, dst, len);
+}
+
+bool
+parse_bool(const char *sval, int *out_status)
+{
+    if (strcmp(sval, "false") == 0) {
```
Comment (mkiiskila) [inline util/parse/src/parse.c]: strcasecmp() would be more user-friendly.

**Comment by ccollins476ad** [comment]
Thanks, Marko.  I agree with all your comments.  I have made the following changes:

* Rename `MN_INET6_MAX_LEN` to `MN_INET6_ADDRSTRLEN`.
* Increase max IPv6 address string length from 39 to 46 (now includes null-terminator).
* Add `#define MN_INET_ADDRSTRLEN  16` to `mn_socket.h`.
* Use `strcasecmp` instead of `strcmp` in `parse_bool`.
* Add `#include "mn_socket/mn_socket.h"` to `parse.c`.

**Pull request merged**